core/app/models/spree/order.rb

```diff
@@ -288,7 +288,12 @@ def allow_cancel?
     end
 
     def all_inventory_units_returned?
-      inventory_units.all?(&:returned?)
+      # Inventory units are transitioned to the return state by ReturnItem
+      # across a different association path goes through CustomerReturn and
+      # ReturnItem instead of using Order#inventory_units. That makes any
+      # the cached association on order stale and require a *reload* before
+      # checking if they're returned.
```

Comment:
I think you could improve the English here a bit, looks like it has been rewritten a few times and the grammar doesn't make much sense anymore. Same goes for the commit.